accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/services/BasicSpringDataRepositoryDatabaseService.java

```diff
@@ -195,7 +195,7 @@ private void checkedInsert(ACCESSION accession, String hash, MODEL model, int ma
 
     private void checkHashDoesNotExist(String hash)
             throws HashAlreadyExistsException {
-        final ACCESSION_ENTITY dbAccession = repository.findOne(hash);
+        final ACCESSION_ENTITY dbAccession = repository.findById(hash).orElse(null);
```

Comment:
I think we could handle the empty Optional instead of forcing this to return null
```
        final Optional<ACCESSION_ENTITY> dbAccession = repository.findById(hash);
        if (dbAccession.isPresent()) {
            throw new HashAlreadyExistsException(dbAccession.get().getHashedMessage(), dbAccession.get().getAccession());
```